Crawler/Analyzers/UtilAnalyzers/SentencesAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Crawler.Configs;
+using Crawler.PartOfSpeechTagger;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Crawler.Analyzers.UtilAnalyzers
+{
+	public class SentencesAnalyzer : ISentencesAnalyzer
+	{
+		private const string PAST_PARTICIPLE_VERB_EXTENDED_TYPE = "VBN";
+		private const string END_OF_SENTENCE_EXTENDED_TYPE = ".";
+
+		private readonly IOptions<DataFilesConfig> config;
+
+		public SentencesAnalyzer(IOptions<DataFilesConfig> dataFilesConfig)
+		{
+			this.config = dataFilesConfig;
+		}
+
+		public float CalculatePassiveVoiceSentencesPercentage(List<PosTagToken> posTagTokens)
+		{
+			var passiveVoiceSentencesCount = 0;
+			var indexOfNextSentenceEnd = posTagTokens.FindIndex(token => token.ExtendedType == END_OF_SENTENCE_EXTENDED_TYPE);
```

Comment:
Where this extended tag is coming from? I don't see it in the clasifier
also why use extended tag when we have out own tag? that is part of the reason you created it

Reply:
> Where this extended tag is coming from?

It came from the original [tag list](https://www.ling.upenn.edu/courses/Fall_2003/ling001/penn_treebank_pos.html), this list contains the original tags that been returned from the Pos-tagger. 

>  I don't see it in the clasifier

This tag is indeed not part of the classifier, but, it's because is not presented in the original tag list. I found it as I worked with the Pos-tagger.  

> why use extended tag when we have out own tag? that is part of the reason you created it

Our tags represent basic "parts of speech" based on high order categories (verbs, nouns, pronouns and such). "VBN" is one of the original tags that represent a "past participle verbs", we don't need it as a category of itself in general, but sometimes we does need to know the more specific tag and therefore I save it as "ExtendedType" .

Reply:
If "." is not in the documentation there might be something fishy with it, how do you know what it contains?

I think that we should not used the extended tag, if we translate it to something of our own we should translate all of it and not use some of the original and some of our tags. maybe the ePosType can be flags and include something like verb and past